xdsl/dialects/riscv.py

```diff
@@ -333,7 +333,13 @@ def assembly_line(self) -> str | None:
 
     @classmethod
     def parse(cls, parser: Parser) -> Self:
-        args = parser.parse_op_args_list()
+        args = (
+            parser.parse_optional_undelimited_comma_separated_list(
+                parser.parse_optional_unresolved_operand,
+                parser.parse_unresolved_operand,
+            )
+            or ()
+        )
```

Comment:
```suggestion
        args = parser.parse_optional_undelimited_comma_separated_list(
            parser.parse_optional_unresolved_operand,
            parser.parse_unresolved_operand,
        )
        if args is None:
            args = ()
```

We try not to exploit the falseyness of `None` due to the ambiguity of double falseyness. Sometimes the result of some function is not `None` but still falsey, which can lead to hard-to-debug behaviour.

Reply:
`parser.parse_optional_undelimited_comma_separated_list` returns either list or `None`. The return value is falsy only if it is empty list or `None` and either case returning an empty tuple is okay.